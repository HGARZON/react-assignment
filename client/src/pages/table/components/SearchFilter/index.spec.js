import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import SeachFilter from './index';

Enzyme.configure({ adapter: new Adapter() })

const data = {
    results: [
        {
            id: 1,
            name: 'Toxic Rick',
            status: 'dead',
            species: 'humanoid',
            origin: {
                name: 'earth',
            },
            location: {
                name: 'Planet 5',
            },
            image: 'https://rickandmortyapi.com/api/character/avatar/361.jpeg',
            episode: [1]
        }
    ]
}

it('renders without crashing', () => {
    shallow(<SeachFilter data={data.results} />);
});

describe('<SeachFilter data={data.results} />', () => {
    /* Snapshot */
    it('renders correctly', () => {
        const component = renderer.create(<SeachFilter data={data.results} />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
});