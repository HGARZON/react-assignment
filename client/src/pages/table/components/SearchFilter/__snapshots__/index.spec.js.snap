// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeachFilter data={data.results} /> renders correctly 1`] = `
<div
  className="characters__container"
>
  <form
    className="search-filter"
  >
    <input
      className="search-filter__field"
      onChange={[Function]}
      placeholder="Enter your favourite character..."
      type="text"
      value=""
    />
  </form>
  <div
    className="character__grid"
  >
    <div
      className="character__details"
    >
      <img
        alt="Toxic Rick"
        className="character__details--image"
        src="https://rickandmortyapi.com/api/character/avatar/361.jpeg"
        title="Toxic Rick"
      />
      <h3
        className="character__title"
      >
        Toxic Rick
      </h3>
      <button
        className="modal__button"
        onClick={[Function]}
      >
        View details
      </button>
    </div>
  </div>
</div>
`;
